framework/src/main/java/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -668,8 +668,13 @@ public TransferResult<V, S> visitTernaryExpression(
     V elseValue = p.getValueOfSubNode(n.getElseOperand());
     V resultValue = null;
     if (thenValue != null && elseValue != null) {
+      // If a conditional expression is a poly expression, then its Java type is the type of its
+      // context. (For example, the type of the condition expression is in `Object o = b ? "" : "";`
+      // is `Object`, not `String`.)
+      // So, use the Java type of the conditional expression and the annotations for each branch.
+      TypeMirror ternaryType = TreeUtils.typeOf(n.getTree());
```

Comment:
It's a bit inconsistent to use "contitional" in the comment but "ternary" in the variable name.